daemon/daemon/swap_out.go

```diff
@@ -0,0 +1,131 @@
+package daemon
+
+import (
+	"bytes"
+	"context"
+	"encoding/hex"
+	"errors"
+	"fmt"
+
+	"github.com/40acres/40swap/daemon/bitcoin"
+	"github.com/40acres/40swap/daemon/database/models"
+	"github.com/40acres/40swap/daemon/swaps"
+	"github.com/btcsuite/btcd/btcec/v2"
+	"github.com/btcsuite/btcd/txscript"
+	log "github.com/sirupsen/logrus"
+)
+
+func (m *SwapMonitor) MonitorSwapOut(ctx context.Context, currentSwap models.SwapOut) error {
+	logger := log.WithField("id", currentSwap.SwapID)
+	logger.Info("processing swap out")
+
+	newSwap, err := m.swapClient.GetSwapOut(ctx, currentSwap.SwapID)
+	switch {
+	case errors.Is(err, swaps.ErrSwapNotFound):
+		logger.Warn("swap not found")
+
+		outcome := models.OutcomeFailed
+		currentSwap.Outcome = &outcome
+		currentSwap.Status = models.StatusDone
+
+		err := m.repository.SaveSwapOut(&currentSwap)
+		if err != nil {
+			return fmt.Errorf("failed to save swap out: %w", err)
+		}
+	case err != nil:
+		return fmt.Errorf("failed to get swap out: %w", err)
+	}
+
+	newStatus := models.SwapStatus(newSwap.Status)
+	changed := currentSwap.Status != newStatus
+	switch newStatus {
+	case models.StatusCreated:
+		logger.Debug("Waiting for payment")
```

Comment:
as per https://github.com/40acres/40swap/pull/56#discussion_r2020723061, logs start in lowercase